.github/workflows/feature-branch.yml

```
@@ -0,0 +1,99 @@
+name: Feature branch CI
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - '**'
+      - '!main'
+      - '!adminV2'
+
+
+env:
+  NODE_VERSION: 20.x
+
+jobs:
+  code-style:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: 'yarn'
+          cache-dependency-path: 'yarn.lock'
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --immutable
+
+      - name: Code style
+        run: |
+          yarn run prettier:check
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: 'yarn'
+          cache-dependency-path: 'yarn.lock'
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --immutable
+
+      - name: Lint
+        run: |
+          yarn run lint
+
+  unit-tests:
```

To have consistent naming between workflows/repos please name it `test`
```suggestion
  test:
```

---

And what if we would like to add different tests? For example integration tests, e2e tests etc? 

I mean.. for now i can change this. No problem. But i want to show that sometimes it is not possible to be consistent between all projects in the company. Consistency despite everything... just to be consistent has drawbacks.

---

and what if not. I want to have solution for current cases not for the future cases.
Because what if we would like to run all the tests in one run, not split between integration and units?

---

>Because what if we would like to run all the tests in one run, not split between integration and units?
It's better to have separate jobs for that. To have quicker feedback. Instead of waiting for all. 
If unit test fail than probably integration test will not run. So you will fix the unit tests and then integration test will fail etc. 

So that is why it's better to have separate jobs for such things.